packages/testcontainers/src/wait-strategies/http-wait-strategy.ts

```diff
@@ -107,9 +126,36 @@ export class HttpWaitStrategy extends AbstractWaitStrategy {
       this.startupTimeout
     );
 
+    if (containerExited) {
```

Comment:
Same here, I think we want to check if `failOnExitedContainer` is true

```suggestion
    if (this.failOnExitedContainer && containerExited) {
```

Reply:
Done!